client/src/main/java/com/alibaba/nacos/client/identify/CredentialWatcher.java

```diff
@@ -39,31 +40,36 @@
     
     private static final Logger SPAS_LOGGER = LogUtils.logger(CredentialWatcher.class);
     
-    private static final long REFRESH_INTERVAL = 10 * 1000;
+    private static final long REFRESH_INTERVAL = 10 * 1000L;
     
     private final CredentialService serviceInstance;
     
     private final String appName;
     
     private String propertyPath;
     
-    private final TimerTask watcher;
-    
     private boolean stopped;
     
+    private final ScheduledThreadPoolExecutor executor;
+    
     @SuppressWarnings("PMD.AvoidUseTimerRule")
     public CredentialWatcher(String appName, CredentialService serviceInstance) {
         this.appName = appName;
         this.serviceInstance = serviceInstance;
         loadCredential(true);
-        watcher = new TimerTask() {
-            private final Timer timer = new Timer(true);
-            
-            private long modified = 0;
-            
-            {
-                timer.schedule(this, REFRESH_INTERVAL, REFRESH_INTERVAL);
+        
+        executor = new ScheduledThreadPoolExecutor(1, new ThreadFactory() {
```

Comment:
you can see here https://github.com/alibaba/nacos/blob/develop/common/src/main/java/com/alibaba/nacos/common/executor/ExecutorFactory.java

Reply:
Thank you.